src/detection_win.cpp

```diff
@@ -328,20 +346,32 @@ void BuildInitialDeviceList() {
 	SP_DEVINFO_DATA* pspDevInfoData = (SP_DEVINFO_DATA*) HeapAlloc(GetProcessHeap(), 0, sizeof(SP_DEVINFO_DATA));
 	pspDevInfoData->cbSize = sizeof(SP_DEVINFO_DATA);
 	for(int i=0; DllSetupDiEnumDeviceInfo(hDevInfo, i, pspDevInfoData); i++) {
-		DWORD nSize=0 ;
-
-		if (!DllSetupDiGetDeviceInstanceId(hDevInfo, pspDevInfoData, buf, sizeof(buf), &nSize)) {
+		if (!DllSetupDiGetDeviceInstanceId(hDevInfo, pspDevInfoData, buf, sizeof(buf), NULL)) {
 			break;
 		}
 
-		DeviceItem_t* item = new DeviceItem_t();
-		item->deviceState = DeviceState_Connect;
-
-		DWORD DataT;
-		DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_HARDWAREID, &DataT, (PBYTE)buf, MAX_PATH, &nSize);
+		PSP_DEVICE_INTERFACE_DETAIL_DATA pspDeviceIntDetail = NULL;
+		if (!GetDeviceDetails(hDevInfo, pspDevInfoData, i, &pspDeviceIntDetail)) {
+			continue;
+		}
```

Comment:
Why is this check needed now?

Reply:
If we are unable to retrieve the device details, then will not be able to obtain any information about the device. If I remember correctly, this was to handle a situation where the device was removed before device information had been acquired.